### PR TITLE
1) YALB-375: Alerts

### PR DESCRIPTION
**Pull request by ModulesUnraveled**
### Description of work
- Creates the alert component

### Testing Link(s)
- [ ] Navigate to the [Alert Story](https://deploy-preview-55--dev-component-library-twig.netlify.app/?path=/story/molecules-alert--alert)
- [ ] Navigate to the [Alert Examples](https://deploy-preview-55--dev-component-library-twig.netlify.app/?path=/story/molecules-alert--alert-examples)

### Functional Review Steps
- [ ] Interact with each of the alert types and verify everything works as expected (including across page reloads)
- [ ] Verify you can install alerts via the CLI

### Design Review
- [ ] Verify the designs match the [Figma Designs](https://www.figma.com/file/OVfmCW5s1gI7xnrM8ibX2y/UI-Kit-Alpha-%5BYaleSites%5D)

### Accessibility Review
- [ ] Verify the component meets Accessibility requirements

### CODE REVIEW

**Comment by ahitson** [comment]
- Change alert titles to medium weight.
- Remove margin under titles when solo, reduce significantly when body text is there.
- Use grey 500 for the bar/divider.
- Center the right-side link/bar/button group vertically within the space.
- At smaller sizes the bar disappears putting the link by the button.
- When link moves under text on smaller sizes, add padding between it and the bottom of the alert box.
- The link was appearing by the chevron at small sizes or still showing when collapsed, should be hidden in the dropdown.
- Can we change the title on collapsed to overflow style (This is a titl...  X)
- The close and chevron need more target space around them. It'll also help balance things out on the smaller sizes.

**Comment by miketullo95** [comment]
everything functions perfectly! 👍  

one comment - The icons are aligned with the block of the text, but on smaller screens, it seems to get unaligned. 

Screen Shot 2022-03-31 at 10 39 18 AM

**Review by kara-franco — CHANGES_REQUESTED** [review]
I love the alerts! Here's my accessibility feedback: 

- [x] `icon__decorative` variable needs to be set to hide icons (triangle, circle, "x" and arrows) from screen readers (currently "image" is being announced). 
- [x] Marketing Alert has an empty button (see image below). Add "Close Alert" to button.
- [ ] Focus indicator for arrows should be styled to match the respective focus indicator for links (see image below). 
- [ ] Second reset button is empty (see image below). 

Marketing Alert: 
Screen Shot 2022-03-31 at 1 38 27 PM

Button Focus: 
Screen Shot 2022-03-31 at 1 41 06 PM

Empty Reset Button: 
Screen Shot 2022-03-31 at 1 43 19 PM

File: components/02-molecules/alert/_alert.scss [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+  align-items: flex-start;
+  padding-right: var(--size-spacing-3);
+
+  @media (min-width: tokens.$break-mobile) {
+    padding-right: var(--size-spacing-7);
+    border-right: var(--border-thickness-1) solid;
+  }
+}
+
+.alert__heading {
```
[the remark this answers is not in the document]
Comment (ModulesUnraveled) [reply components/02-molecules/alert/_alert.scss:65]: Ashley had the same feedback. It's fixed now.

**Comment by ModulesUnraveled** [comment]
@ahitson I think I've addressed everything according to our zoom call yesterday.

@miketullo95 On small screens, everything is top-aligned, so the text appears a little off from the icon because of the line-height of the text. On large screens everything is centered, so things may or may not line up depending on the content (see screenshot below)

@kara-franco Thanks for catching all of that. The `icon__decorative` thing almost makes me wonder if we should enable that by default...

Screen Shot 2022-04-01 at 5 39 14 AM

**Comment by kara-franco** [comment]
> https://deploy-preview-55--dev-component-library-twig.netlify.app

I agree, I think 95% of the time an icon will be decorative!

**Comment by kara-franco** [comment]
Hmm, something is funky on my end when looking at Storybook: 

Screen Shot 2022-04-01 at 1 36 12 PM

**Comment by ModulesUnraveled** [comment]
@kara-franco That's a Netlify issue (that we should probably figure out). Basically, I added some new tokens to the tokens repo after the initial build in Netlify. Netlify tries to save time by using a cached `node_modules` directory, so it didn't get the new tokens. I just clicked the button to rebuild without cache, so it should look correct in a minute or so

**Comment by miketullo95** [comment]
> @miketullo95 On small screens, everything is top-aligned, so the text appears a little off from the icon because of the line-height of the text. On large screens everything is centered, so things may or may not line up depending on the content (see screenshot below)

Thanks for the info! Would it be possible to change the alignment on mobile so that the icon is center-aligned to the h2 first line text? On certain sizes its reads a little like a superscript. Let me know if that made sense lol. 

Thanks! 😄

**Review by kara-franco — APPROVED** [review]
Excellent, all fixes look great!

**Comment by ModulesUnraveled** [comment]
@miketullo95 I worked with Ashley on the icon alignment (for both mobile and desktop) please check again.

@ahitson Please check as well

**Review by miketullo95 — APPROVED** [review]
icon looks great! 👍

**Pull request merged**